agent/acs/session/task_stop_verification_ack_responder.go

```diff
@@ -39,6 +39,7 @@ func (ts *taskStopper) StopTask(taskARN string) {
 			loggerfield.TaskARN: task.Arn,
 		})
 		task.SetDesiredStatus(apitaskstatus.TaskStopped)
+		task.UpdateDesiredStatus()
 		ts.taskEngine.AddTask(task)
 	} else {
```

Comment:
Saving to disk shouldn't necessary in this case since desired status is STOPPED.

For reference, see: https://github.com/aws/amazon-ecs-agent/blob/master/agent/acs/session/payload_responder.go#L254-L256

Reply:
UPDATE: Based on above discussions and since we are aligned call to AddTask is being removed, we will add logic to save task state here as part of this pull request. See https://github.com/aws/amazon-ecs-agent/pull/4240#discussion_r1674534285 for reference.